symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/signal/SignalService.java

```diff
@@ -0,0 +1,162 @@
+package com.symphony.bdk.core.service.signal;
+
+import com.symphony.bdk.core.auth.AuthSession;
+import com.symphony.bdk.core.retry.RetryWithRecovery;
+import com.symphony.bdk.core.retry.RetryWithRecoveryBuilder;
+import com.symphony.bdk.core.service.OboService;
+import com.symphony.bdk.core.service.pagination.PaginatedApi;
+import com.symphony.bdk.core.service.pagination.PaginatedService;
+import com.symphony.bdk.core.util.function.SupplierWithApiException;
+import com.symphony.bdk.gen.api.SignalsApi;
+import com.symphony.bdk.gen.api.model.BaseSignal;
+import com.symphony.bdk.gen.api.model.ChannelSubscriber;
+import com.symphony.bdk.gen.api.model.ChannelSubscriptionResponse;
+import com.symphony.bdk.gen.api.model.Signal;
+
+import com.symphony.bdk.http.api.ApiException;
+
+import org.apiguardian.api.API;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+/**
+ * Service class for managing signal information.
+ * <p>
+ * This service is used for listing signals related to the user, get information of a specified signal
+ * or perform some actions related to the signal like:
+ * <p><ul>
+ * <li>Create a signal</li>
+ * <li>Update a signal</li>
+ * <li>Delete a signal</li>
+ * <li>Subscribe or unsubscribe a signal</li>
+ * </ul></p>
+ */
+@API(status = API.Status.STABLE)
+public class SignalService implements OboSignalService, OboService<OboSignalService> {
+
+  private final SignalsApi signalsApi;
+  private final AuthSession authSession;
+  private final RetryWithRecoveryBuilder<?> retryBuilder;
+
+  public SignalService(SignalsApi signalsApi, AuthSession authSession, RetryWithRecoveryBuilder<?> retryBuilder) {
+    this.signalsApi = signalsApi;
+    this.authSession = authSession;
+    this.retryBuilder = retryBuilder;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public OboSignalService obo(AuthSession oboSession) {
+    return new SignalService(signalsApi, oboSession, retryBuilder);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public List<Signal> listSignals(Integer skip, Integer limit) {
+    return executeAndRetry("listSignals",
+        () -> signalsApi.v1SignalsListGet(authSession.getSessionToken(), authSession.getKeyManagerToken(), skip, limit));
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Stream<Signal> listSignalsStream(Integer chunkSize, Integer totalSize) {
+    PaginatedApi<Signal> api = ((this::listSignals));
+
+    final int actualChunkSize = chunkSize == null ? 50 : chunkSize;
+    final int actualTotalSize = totalSize == null ? 50 : totalSize;
```

Comment:
Why we don't have the same checks in `listSignals(Integer skip, Integer limit)`?

Reply:
because we can pass null value to the SignalsApi#v1SignalsListGet to make a call to endpoint without specified skip and limit. The default value will be used in agent side.